### PR TITLE
Release 2.22.0 (UPLOAD-106, UPLOAD-116 and UPLOAD-117)

### PR DESCRIPTION
**Pull request by gniezen**
Addresses:
- [UPLOAD-106]
- [UPLOAD-116]
- [UPLOAD-117]

[UPLOAD-106]: https://tidepool.atlassian.net/browse/UPLOAD-106
[UPLOAD-116]: https://tidepool.atlassian.net/browse/UPLOAD-116
[UPLOAD-117]: https://tidepool.atlassian.net/browse/UPLOAD-117

### CODE REVIEW

**Comment by gniezen** [comment]
Hi there @pazaan ! Would you mind reviewing the changes since your last code review?

**Comment by gniezen** [comment]
Hmm, turns out @eluhrs found some more issues - hopefully these will be the final changes to this release. :crossed_fingers:

**Comment by gniezen** [comment]
@pazaan Would you mind having a look at the changes since the last review, as that's the last thing before I can push v.2.22.0 to prod?

File: lib/drivers/abbott/abbottFreeStyleLite.js [hunk 1]
```diff
@@ -357,30 +357,21 @@ module.exports = function (config) {
...
       if (err) {
-        debug('Failure trying to talk to device.');
-        debug(err);
-        debug(result);
         cb(err, null);
-      } else {
-        debug('Device time set');
-        cb(null);
       }
+      console.log('Device time set');
```
Comment (pazaan) [inline lib/drivers/abbott/abbottFreeStyleLite.js]: Should this be `debug`?

File: lib/drivers/onetouch/oneTouchUltra2.js [hunk 2]
```diff
@@ -179,6 +179,110 @@ module.exports = function (config) {
...
+            return callback();
+          });
+        },
+        function (err, n) {
+          if (err) {
+            return cb(err);
+          }
+
+          // get data from lines
+          var dpattern = /P "(\w+)","(\d{2}\/\d{2}\/\d{2})","(\d{2}:\d{2}:\d{2})   ","(..\d{3} )","(.)","(\d{2})",\W00\W(.+)/;
```
Comment (pazaan) [inline lib/drivers/onetouch/oneTouchUltra2.js]: What are the extra `..` characters in `(..\d{3} )`? `(.)` is very inclusive. Can it include whitespace characters?

Comment (gniezen) [reply lib/drivers/onetouch/oneTouchUltra2.js]: I copied & pasted this regex from another section in the Ultra2 driver - we've been using it for years without any problems. TBH it's one of those where I can't even remember who actually wrote the code.

Looking at my regex cheatsheet `(.)` can include any character except for newlines.

File: lib/drivers/abbott/abbottFreeStyleLite.js (hunk 1 above)
Comment (gniezen) [reply lib/drivers/abbott/abbottFreeStyleLite.js]: Yes, you're right. I'm not going to make the change the right now, as that would mean another re-build and QA before we can push to prod, and this is not user-facing. But I will fix it in another branch to get merged into `develop`.

**Pull request merged**